### DISCUSSION TITLE
How to use rpcpooling option

### DISCUSSION THREAD

**Discussion by kannapoix** [lightningnetwork/lnd#9067] [Q&A] [answered]
LND has an `rpcpolling` option to fetch transactions from bitcoind via RPC instead of ZMQ. I'm having trouble using this option and getting the following error:

>failed to load config: ValidateConfig: unable to load RPC credentials for bitcoind: unable to extract RPC credentials: unable to find zmqpubrawblock in config, cannot start w/o RPC connection

Are there any other options that need to be added along with `rpcpolling`? I thought if I used `rpcpolling`, I wouldn't need to set zmqpubrawblock. 
When I explicitly add `bitcoind.rpccookie`, LND starts properly. Without `bitcoind.rpccookie`, LND loads the cookie from the default location after checking for the existence of ZMQ parameters, which might be the issue.

**Comment by guggero** [comment]
The error message is a bit misleading. But basically the part that's relevant is `unable to extract RPC credentials`. It means you don't have the authentication set. So you must set either `bitcoind.rpcuser` and `bitcoind.rpcpass`, _or_ `bitcoind.rpccookie`.

**Reply by kannapoix** [reply]
@guggero Thanks for your quick reply. I thought that LND would load the cookie file from the default path even if I didn't set any of the options you mentioned. Is that not the case? Or the loading from the default path is failing in this case.

**Reply by guggero** [reply, marked as answer]
I think the logic that loads the cookie from the default path wasn't updated for the `rpcpolling` option, so it complains about the missing ZMQ settings. Which I guess is a bug. But that logic is very old and in need of a refactor anyway... You can avoid that by manually specifying the cookie or user/password option.

**Reply by kannapoix** [reply]
OK, I get it. Thank you!
